### PR TITLE
Expose a few BWA backtrack options.

### PR DESCRIPTION
**Pull request by nh13**
I could change `maxDiffOrMissingProb` to two `Options` and make sure both are not `Some` value, but I went with `Either` since it seemed more "scala".

### CODE REVIEW

File: tasks/src/main/scala/dagr/tasks/bwa/BwaBacktrack.scala [hunk 1]
```diff
@@ -16,12 +16,17 @@ import scala.collection.mutable.ListBuffer
   * BAM file via bwa backtrack/bwa short alignment.
   */
 class BwaBacktrack(val unmappedBam : PathToBam, val mappedBam : PathToBam,
-                   val ref : PathToFasta, val tmpDir: Option[DirPath] = None
+                   val ref : PathToFasta, val tmpDir: Option[DirPath] = None,
+                   val maxDiffOrMissingProb: Option[Either[Int, Float]] = None,
```
Comment (tfenne) [inline tasks/src/main/scala/dagr/tasks/bwa/BwaBacktrack.scala:20]: It's really annoying that scala's numeric types don't have a common superclass, so you can't just do `Option[Number]`

File: tasks/src/main/scala/dagr/tasks/bwa/BwaBacktrack.scala [hunk 2]
```diff
@@ -16,12 +16,17 @@ import scala.collection.mutable.ListBuffer
...
+                   val seedLength: Option[Int] = None,
+                   val pairedEnd: Boolean = true
                 ) extends Pipeline {
   /**
     * Strings together invocations of bwa aln and bwa sampe to generate an output SAM file.
     */
-  override def build(): Unit = {
+  override def build(): Unit = if (pairedEnd) buildPaired() else buildFragment()
+
+  def buildPaired(): Unit = {
```
Comment (tfenne) [inline tasks/src/main/scala/dagr/tasks/bwa/BwaBacktrack.scala]: This should probably be private.

**Comment by tfenne** [comment]
Looks reasonable to me.  Maybe make the build\* methods private?  Merge at will.

File: tasks/src/main/scala/dagr/tasks/bwa/BwaBacktrack.scala (hunk 1 above)
Comment (nh13) [inline tasks/src/main/scala/dagr/tasks/bwa/BwaBacktrack.scala:20]: My impression on reading stackoverflow is most people would just say "roll your own", which tells me they don't get it.

File: tasks/src/main/scala/dagr/tasks/bwa/BwaBacktrack.scala (hunk 2 above)
Comment (nh13) [inline tasks/src/main/scala/dagr/tasks/bwa/BwaBacktrack.scala]: Good catch.

**Pull request merged**